cmd/release-controller/sync_tags.go

```diff
@@ -215,6 +215,44 @@ func (c *Controller) ensureReleaseTagPhase(release *Release, preconditionPhases
 	return nil
 }
 
+// Update annotations for a set of tags in a release
+func (c *Controller) updateReleaseTagAnnotations(release *Release, tagAnnotations map[string]map[string]string) error {
+	if len(tagAnnotations) == 0 {
+		return nil
+	}
+	var changed bool
+	target := release.Target.DeepCopy()
+	for name, annotations := range tagAnnotations {
+		tag := findTagReference(target, name)
+		if tag == nil {
+			return fmt.Errorf("release %s no longer exists, cannot update annotations", name)
+		}
+		if tag.Annotations == nil {
+			tag.Annotations = make(map[string]string)
+		}
+		for k, v := range annotations {
+			if _, ok := tag.Annotations[k]; ok && len(v) == 0 {
```

Comment:
generally don't use this structure, use `continue` and successive if blocks rather than else.